crates/driver/src/boundary/settlement.rs

```diff
@@ -57,8 +56,8 @@ pub struct Settlement {
 impl Settlement {
     pub async fn encode(
         eth: &Ethereum,
-        solver: &Solver,
         solution: &competition::Solution,
+        // TODO I think it's possible to remove this parameter, do this in a follow-up
```

Comment:
I think it is fine for the `competition::Solution` domain object to include enough information for encoding a settlement.